File: include/apl/errors/SyntaxError.java
package APL.errors;

import APL.types.*;

public class SyntaxError extends APLError {
  public SyntaxError(std::string s) {
    super(s);
  }
  public SyntaxError(std::string s, Tokenable fun) {
    super(s, fun);
  }
  public SyntaxError(std::string s, Callable fun, Tokenable cause) {
    super(s, fun, cause);
  }

  public static void must(bool b, std::string msg) {
    if (!b) throw new SyntaxError(msg);
  }
}

File: APL/errors/SyntaxError.cpp
#include <APL/SyntaxError.h>

namespace APL::errors
{
    // using namespace APL::types;

    SyntaxError::SyntaxError(std::string const& s) : APLError(s)
    {
    }

    SyntaxError::SyntaxError(std::string const& s, std::shared_ptr<APL::types::Tokenable> fun) : APLError(s, fun)
    {
    }

    SyntaxError::SyntaxError(std::string const& s, std::shared_ptr<APL::types::Callable> fun, std::shared_ptr<APL::types::Tokenable> cause) : APLError(s, fun, cause)
    {
    }

    void SyntaxError::must(bool b, std::string const& msg)
    {
      if (!b)
      {
          throw SyntaxError(StringHelper::wstring_to_string(msg));
      }
    }
}
